app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,69 @@
+class DriversController < ApplicationController
+  def index
+    @drivers = Driver.all
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+
+    if @driver.save
+      redirect_to driver_path(@driver.id)
+    else
+    render :new
+    end
+  end
+
+  def show
+    driver_id = params[:id].to_i
+    @driver = Driver.find_by(id: driver_id)
+    #
+    if @driver.nil?
+      head :not_found
+    end
+  end
+
+  def edit
+    @driver = Driver.find_by(id: params[:id])
+    if @driver.nil?
```

Comment:
Similar to above, now that we know the power of redirects, in these cases, it might be best to redirect to a page like the homepage, in case the driver isn't found.